exporter/datadogexporter/config.go

```diff
@@ -0,0 +1,549 @@
+}
+
+// LimitedTLSClientSetting is a subset of TLSClientSetting, see LimitedHTTPClientSettings for more details
+type LimitedTLSClientSettings struct {
+	// InsecureSkipVerify controls whether a client verifies the server's
+	// certificate chain and host name.
+	InsecureSkipVerify bool `mapstructure:"insecure_skip_verify"`
+}
+
+type LimitedHTTPClientSettings struct {
+	TLSSetting LimitedTLSClientSettings `mapstructure:"tls,omitempty"`
+}
+
+// Config defines configuration for the Datadog exporter.
+type Config struct {
+	config.ExporterSettings        `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
+
+	LimitedHTTPClientSettings `mapstructure:",squash"`
+
+	TagsConfig `mapstructure:",squash"`
+
+	// API defines the Datadog API configuration.
+	API APIConfig `mapstructure:"api"`
+
+	// Metrics defines the Metrics exporter specific configuration
+	Metrics MetricsConfig `mapstructure:"metrics"`
+
+	// Traces defines the Traces exporter specific configuration
+	Traces TracesConfig `mapstructure:"traces"`
+
+	// HostMetadata defines the host metadata specific configuration
+	HostMetadata HostMetadataConfig `mapstructure:"host_metadata"`
+
+	// SendMetadata defines whether to send host metadata
+	// This is undocumented and only used for unit testing.
+	//
+	// This can't be disabled if `only_metadata` is true.
+	// Deprecated: [v0.49.0] Use `host_metadata::enabled` (HostMetadata.Enabled) instead.
+	SendMetadata bool `mapstructure:"send_metadata"`
+
+	// OnlyMetadata defines whether to only send metadata
+	// This is useful for agent-collector setups, so that
+	// metadata about a host is sent to the backend even
+	// when telemetry data is reported via a different host.
+	//
+	// This flag is incompatible with disabling host metadata,
+	// `use_resource_metadata`, or `host_metadata::hostname_source != first_resource`
+	OnlyMetadata bool `mapstructure:"only_metadata"`
+
+	// UseResourceMetadata defines whether to use resource attributes
+	// for completing host metadata (such as the hostname or host tags).
+	//
+	// By default this is true: the first resource attribute getting to
+	// the exporter will be used for host metadata.
+	// Disable this in the Collector if you are using an agent-collector setup.
+	// Deprecated: [v0.49.0] Use `host_metadata::hostname_source` (HostMetadata.HostnameSource) instead.
+	UseResourceMetadata bool `mapstructure:"use_resource_metadata"`
+
+	// warnings stores non-fatal configuration errors.
+	warnings []error
+}
+
+// logWarnings logs warning messages that were generated on unmarshaling.
+func (c *Config) logWarnings(logger *zap.Logger) {
+	for _, err := range c.warnings {
+		logger.Warn(fmt.Sprintf("%v", err))
+	}
+}
```

Comment:
This file was copied over from the `config` package. The only change is on this function (done on the second commit): `Sanitize` was renamed to `logWarnings`, making it unexported.

Reply:
Can we not log those in `Validate()`?